internal/insights/job.go

```diff
@@ -0,0 +1,184 @@
+package insights
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/saucelabs/saucectl/internal/job"
+	"github.com/saucelabs/saucectl/internal/requesth"
+)
+
+const (
+	RDCSource = "rdc"
+	VDCSource = "vdc"
+	APISource = "api"
+)
+
+// ListJobResp represents list job response structure
+type ListJobResp struct {
+	Jobs  []JobResp `json:"jobs"`
+	Total int       `json:"total"`
+}
+
+// JobResp represents job response structure
+type JobResp struct {
+	ID          string `json:"id"`
+	Name        string `json:"name"`
+	Status      string `json:"status"`
+	Error       string `json:"error"`
+	Framework   string `json:"automation_backend"`
+	Device      string `json:"device"`
+	BrowserName string `json:"browser_name"`
+	OS          string `json:"os"`
+	OSVersion   string `json:"os_version"`
+	Source      string `json:"source"`
+}
+
+const AutomaticRunMode = "automatic"
```

Comment:
Would be good to explain what this thing is/does 🙏 I have a guess as to what it does, but I'd rather not guess 😅

Reply:
sure! I'll add a comment